src/services/NurseryWithdrawalService.ts

```diff
@@ -215,6 +216,55 @@ const getWithdrawalPhotosList = async (withdrawalId: number): Promise<Response &
   return response;
 };
 
+/**
+ * Get Filter Options
+ */
+const getFilterOptions = async (organizationId: number): Promise<FieldOptionsMap> => {
+  const searchParams: SearchRequestPayload = {
+    prefix: 'nurseryWithdrawals',
+    fields: [
+      'id',
+      'purpose',
+      'facility_name',
+      'destinationName',
+      'plotNames',
+      'batchWithdrawals.batch_species_scientificName',
+    ],
+    search: SearchService.convertToSearchNodePayload([], organizationId),
+    sortOrder: [{ field: 'id', direction: 'Ascending' }],
+    count: 1000,
+  };
+
+  const data = await SearchService.search(searchParams);
+  const map: FieldOptionsMap = {};
+  if (data) {
```

Comment:
This part feels messy. I had to get the result into a form that the FilterGroup expected. I may refactor that into something cleaner / more generic.

Reply:
This is fine.. if you want a more functional approach/suggestion.. here's something that may work?

```
const map: FieldOptionsMap = (data ?? []).reduce((resultAccumulator, datum) => {
  return Object.keys(datum).reduce((accumulator, key) => {
    const isBatchWithdrawals = key === 'batchWithdrawals';
    const k = isBatchWithdrawals ? 'batchWithdrawals.batch_species_scientificName' : key;
    const value = isBatchWithdrawals   
      ? (datum[k] as any[]).map(batchWithdrawl => batchWithdrawal.batch_species_scientificName)
      : datum[k] as string
      
    if (!accumulator[k]) {
      accumulator[k] = { partial: false, values: [] };
    }
    accumulator[k].values.push( isArray(value) ? ...value : value);
      
    return accumulator;
  }, resultAccumulator);
}, {});

```